test/index.test.js

```diff
@@ -191,6 +194,33 @@ describe('index test', () => {
                 done();
             });
         });
+
+        it('will short circuit retries when special conditions met', (done) => {
+            breaker = new Breaker('testFn', {
+                shouldRetry: (err, args) => args[0] === '1' && err.message !== 'request failure2',
+                breaker: {
+                    resetTimeout: 1000
+                },
+                retry: {
+                    minTimeout: 25
+                }
+            });
+
+            const error = new Error('request failure');
+            const error2 = new Error('request failure2');
+
+            breakerMock.rejects(error);
+            breakerMock.onThirdCall().rejects(error2);
+
+            breaker.runCommand('1', '2', (err, data) => {
+                assert.notOk(data);
+                assert.deepEqual(err, error);
+                assert.callCount(breakerMock, 3);
+                assert.equal(breaker.isClosed(), true);
+                assert.equal(breaker.getFailedRequests(), 4);
```

Comment:
It turns out that `circuitbreaker` is mocked out entirely by the tests, and is not leveraged at all. The fact that we return 4 is only because that is what is mocked to return. This makes this a bogus check.